tests/Doctrine/Tests/ORM/Tools/SchemaToolTest.php

```diff
@@ -86,6 +88,44 @@ public function testPassColumnDefinitionToJoinColumn()
         $this->assertEquals($customColumnDef, $table->getColumn('avatar_id')->getColumnDefinition());
     }
 
+    public function testPassColumnOptionsToJoinColumn()
+    {
+        $em = $this->_getTestEntityManager();
+        $schemaTool = new SchemaTool($em);
+
+        $category = $em->getClassMetadata(ForumCategory::class);
+        $board = $em->getClassMetadata(ForumBoard::class);
+
+        $classes = array($category, $board);
+
+        $schema = $schemaTool->getSchemaFromMetadata($classes);
+
+        $this->assertTrue($schema->hasTable('forum_categories'));
+        $this->assertTrue($schema->hasTable('forum_boards'));
+
+        $tableCategory = $schema->getTable('forum_categories');
+        $tableBoard = $schema->getTable('forum_boards');
+
+        $this->assertTrue($tableBoard->hasColumn('category_id'));
+
+        $this->assertSame(
+         $tableCategory->getColumn('id')->getFixed(),
+         $tableBoard->getColumn('category_id')->getFixed(),
+         'Foreign key/join column should have the same value of option `fixed` as the referenced column'
+        );
+
+        $this->assertEquals(
+         $tableCategory->getColumn('id')->getCustomSchemaOptions(),
+         $tableBoard->getColumn('category_id')->getCustomSchemaOptions(),
+         'Foreign key/join column should have the same custom options as the referenced column'
+        );
+
+        $this->assertNotEmpty(
```

Comment:
This should probably assert the options themselves, since the test is called _pass column options to joinColumn_ but doesn't actually check for custom options being where they should be.